File: user-service/src/main/java/rs/edu/raf/repository/EmployeeRepository.java
package rs.edu.raf.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import rs.edu.raf.model.user.Employee;

import java.util.List;
import java.util.Optional;

@Repository
public interface EmployeeRepository extends JpaRepository<Employee, Long> {
    List<Employee> findAllByActiveIsTrue();
    Optional<Employee> findByEmailAndActiveIsTrue(String email);
}

File: user-service-kotlin/src/main/kotlin/org/example/userservicekotlin/repository/EmployeeRepository.kt
package org.example.userservicekotlin.repository

import org.example.userservicekotlin.model.user.Employee
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface EmployeeRepository : JpaRepository<Employee, Long> {
    fun findAllByActiveIsTrue(): List<Employee>
    fun findByEmailAndActiveIsTrue(email: String): Optional<Employee>
}